README.txt
cbrealurl README

This extension aims to help configure the realurl extension for multiple
extensions without much input.

@author Michael Cannon <michael@cannonbose.com>
@version $Id: README.txt,v 1.6 2008/04/02 22:44:45 cannon Exp $

Read the manual at doc/manual.sxw


CHANGELOG
2008-04-02 Wednesday 18:50  cannon

	* ext_emconf.php: post 1.1.0 checkin

2008-04-02 Wednesday 18:49  cannon

	* ext_emconf.php, doc/manual.sxw: manual title revision

2008-04-02 Wednesday 18:46  cannon

	* ext_emconf.php, CHANGELOG: post 1.0.4 checkin

2008-04-02 Wednesday 18:44  cannon

	* CHANGELOG, README.txt: revise readme to manual

2008-04-02 Wednesday 18:44  cannon

	* doc/manual.sxw: Inital import of "manual.sxw" file

2008-04-02 Wednesday 17:26  cannon

	* realurl.php: custom var

2008-04-02 Wednesday 16:02  cannon

	* static/CB_RealURL/constants.txt: create cbrealurl page

2008-04-02 Wednesday 15:59  cannon

	* README.txt, realurl.php, static/CB_RealURL/constants.txt: create
	  own constants pages; drop realurl-custom.php in ext linking

2008-04-02 Wednesday 15:57  cannon

	* realurl-custom-sample.php, realurl-custom.php: Move
	  "realurl-custom.php" to "realurl-custom-sample.php"

2007-11-29 Thursday 18:26  cannon

	* CHANGELOG, ext_emconf.php: 1.0.3 checkin

2007-11-29 Thursday 18:23  cannon

	* realurl.php: mod default html and 404 erroring

2007-11-29 Thursday 18:16  cannon

	* CHANGELOG: Inital import of "CHANGELOG" file

2007-11-29 Thursday 18:15  cannon

	* ChangeLog: Remoal of "ChangeLog"

2007-11-29 Thursday 18:13  cannon

	* realurl.php: update realurl to be more like autoconf from
	  realurl; backPid key

2007-11-23 Friday 08:54  cannon

	* ext_emconf.php: misc category

2007-11-22 Thursday 11:27  cannon

	* ChangeLog, ext_emconf.php: 1.0.2 checkin

2007-11-22 Thursday 11:24  cannon

	* README.txt, realurl-custom.php, realurl.php: README update;
	  realurl conf cleanup

2007-11-22 Thursday 10:54  cannon

	* ChangeLog, ext_emconf.php: version 1.0.1 checkin

2007-11-22 Thursday 10:48  cannon

	* README.txt, realurl-custom.php, realurl.php: README update,
	  realurl config change, dropping of baseline rootpage_id

2007-11-21 Wednesday 14:28  cannon

	* README.txt: update readme with short install

2007-11-21 Wednesday 14:18  cannon

	* ext_emconf.php: 1.0.0 check-in

2007-11-21 Wednesday 14:15  cannon

	* realurl.php: narc/ncat change

2007-11-21 Wednesday 14:12  cannon

	* ChangeLog, README.txt, ext_emconf.php, ext_icon.gif,
	  ext_localconf.php, ext_tables.php, realurl-custom.php,
	  realurl.php, doc/wizard_form.dat, doc/wizard_form.html,
	  static/CB_RealURL/constants.txt, static/CB_RealURL/setup.txt
	  (utags: initial): Initial cbrealurl TYPO3 extension import

2007-11-21 Wednesday 14:12  cannon

	* ChangeLog, README.txt, ext_emconf.php, ext_icon.gif,
	  ext_localconf.php, ext_tables.php, realurl-custom.php,
	  realurl.php, doc/wizard_form.dat, doc/wizard_form.html,
	  static/CB_RealURL/constants.txt, static/CB_RealURL/setup.txt:
	  Initial revision



realurl.php
<?php

/**
 * RealURL helper script
 *
 * @ref http://typo3bloke.net/post-details/archive/2008/may/30/realurl_made_easy_part_1/
 *
 * @author Michael Cannon <michael@cannonbose.com>
 * @version $Id: realurl.php,v 1.13 2008/12/10 11:07:09 cannon Exp $
 */

// Prevent new realurl updates from clearing cache
unset($TYPO3_CONF_VARS['SC_OPTIONS']['t3lib/class.t3lib_tcemain.php']['clearAllCache_additionalTables']['tx_realurl_pathcache']);

// realurl naming precedence configuration
$TYPO3_CONF_VARS['FE']['addRootLineFields'] .= ',tx_realurl_pathsegment,alias,title';

$TYPO3_CONF_VARS['EXTCONF']['realurl'] = array();
$TYPO3_CONF_VARS['EXTCONF']['realurl']['_DEFAULT'] = array(
	'init' => array( 
		'enableCHashCache' => true
		, 'appendMissingSlash' => 'ifNotFile'
		, 'adminJumpToBackend' => true
		, 'enableUrlDecodeCache' => true
		, 'enableUrlEncodeCache' => true
		, 'emptyUrlReturnValue' => '/'
		// Allow for proper SEO 404 handling
		, 'postVarSet_failureMode' => ''
	)   
	, 'redirects'		=> array()
	, 'preVars' => array(
			array(   
				'GETvar' => 'no_cache'
				, 'valueMap' => array( 
					'nc' => 1
			)
			, 'noMatch' => 'bypass'
		)
		, array(   
			'GETvar' => 'L', 
			'valueMap' => array(
				// id's need to line up with Website Language Ids in TYPO3
				// 'english' => '0',
				'en' => '0',
				'us' => '0',
				// 'french' => '1',
				'fr' => '1',
				// Simplified Chinese
				'cn' => '2',
				// norwegian?
				// 'no' => '2',
				// Simplified Chinese
				'tw' => '3',
				// 'german' => '3',
				// 'de' => '3',
				// 'danish' => '4',
				'dk' => '4',
				// 'spanish' => '5',
				'es' => '5',
				// British English
				'uk' => '6',
				// Traditional Chinese
				'tw' => '8',
			),   
			'noMatch' => 'bypass',
		),
	)
	, 'pagePath' => array(
		'type'			=> 'user'
		, 'userFunc' => 'EXT:realurl/class.tx_realurl_advanced.php:&tx_realurl_advanced->main'
		, 'spaceCharacter'	=> '-'
		, 'languageGetVar'	=> 'L'
		, 'rootpage_id'		=> 1
		, 'segTitleFieldList'	=> 'tx_realurl_pathsegment,alias,title'
	)
	, 'fixedPostVars'	=> array()
	, 'postVarSets' => array(
		'_DEFAULT' => array(
			// news archive parameters
			'archive' => array(
				array(
					'GETvar' => 'tx_ttnews[year]'
				)
				, array(
					'GETvar' => 'tx_ttnews[month]'
					// MLC uncomment if month names instead of numbers are
					// desired
					/*
					, 'valueMap' => array(
						'january' => '01',
						'february' => '02',
						'march' => '03', 
						'april' => '04',
						'may' => '05',
						'june' => '06',
						'july' => '07',
						'august' => '08',
						'september' => '09',
						'october' => '10',
						'november' => '11',
						'december' => '12',
					)
					*/
				)
				, array(
					'GETvar' => 'tx_ttnews[day]'
					, 'noMatch' => 'bypass',
				)
				, array(
					'GETvar' => 'tx_ttnews[pS]'
					, 'noMatch' => 'bypass',
				)
				, array(
					'GETvar' => 'tx_ttnews[pL]'
					, 'noMatch' => 'bypass',
				)
			)
			// news pagebrowser
			, 'p' => array(
				array(
					'GETvar' => 'tx_ttnews[pointer]'
				)
			)
			// news category
			, 'c' => array (
				array(
					'GETvar' => 'tx_ttnews[cat]'
					, 'lookUpTable' => array(
						'table' => 'tt_news_cat'
						, 'id_field' => 'uid'
						, 'alias_field' => 'title'
						, 'addWhereClause' => ' AND deleted != 1'
						, 'useUniqueCache' => 1
						, 'useUniqueCache_conf' => array(
							'strtolower' => 1
						),
					),
				),
			)
			// news item
			, 'article' => array(
				array(
					'GETvar' => 'tx_ttnews[tt_news]'
					, 'lookUpTable' => array(
						'table' => 'tt_news'
						, 'id_field' => 'uid'
						// MLC Goolge wants uniqueness for spidering
						, 'alias_field' => 'concat(title, " ", uid)'
						, 'addWhereClause' => ' AND deleted != 1'
						, 'useUniqueCache' => 1
						, 'useUniqueCache_conf' => array(
							'strtolower' => 1
							, 'spaceCharacter' => '-'
						)
					)
				)
				, array(
					'GETvar' => 'tx_ttnews[swords]'
				)
			)
			, 'abp' => array(
				array(
					'GETvar' => 'tx_ttnews[backPid]'
				)
			)
			, 'nq' => array(
				array(
					'GETvar' => 'news_search[search_text]'
				)
			)
			, 'nqc' => array(
				array(
					'GETvar' => 'news_search[category][]'
				)
			)
			, 'login'	=> array(
				array(
					'GETvar'	=> 'tx_newloginbox_pi3[showUid]'
				)
			)
			, 'forgot-login'	=> array(
				array(
					'GETvar'	=> 'tx_newloginbox_pi1[forgot]'
				)
			)
			, 'forgot'	=> array(
				array(
					'GETvar'	=> 'tx_felogin_pi1[forgot]'
				)
			)
			, 'query'	=> array(
				array(
					'GETvar'	=> 'tx_indexedsearch[sword]'
				)
				, array(
					'GETvar'	=> 'tx_indexedsearch[ext]'
				)
				, array(
					'GETvar'	=> 'tx_indexedsearch[submit_button]'
				)
				, array(
					'GETvar'	=> 'tx_indexedsearch[_sections]'
				)
				, array(
					'GETvar'	=> 'tx_indexedsearch[pointer]'
				)
				, array(
					'GETvar'	=> 'tx_indexedsearch[extResume]'
				)
				, array(
					'GETvar'	=> 'tx_indexedsearch[type]'
				)
				, array(
					'GETvar'	=> 'tx_indexedsearch[group]'
				)
				, array(
					'GETvar'	=> 'tx_indexedsearch[_freeIndexUid]'
				)
				, array(
					'GETvar'	=> 'tx_indexedsearch[media]'
				)
				, array(
					'GETvar'	=> 'tx_indexedsearch[defOp]'
				)
				, array(
					'GETvar'	=> 'tx_indexedsearch[ang]'
				)
				, array(
					'GETvar'	=> 'tx_indexedsearch[desc]'
				)
				, array(
					'GETvar'	=> 'tx_indexedsearch[results]'
				)
				, array(
					'GETvar'	=> 'tx_indexedsearch[sections]'
				)
				, array(
					'GETvar'	=> 'tx_indexedsearch[lang]'
				)
				, array(
					'GETvar'	=> 'tx_indexedsearch[order]'
				)
				, array(
					'GETvar'	=> 'tx_indexedsearch[freeIndexUid]'
				)
			)
			, 'srfu'	=> array(
				array(
					'GETvar'	=> 'tx_srfeuserregister_pi1[cmd]'
				)
				, array(
					'GETvar'	=> 'tx_srfeuserregister_pi1[pointer]'
				)
				, array(
					'GETvar'	=> 'tx_srfeuserregister_pi1[mode]'
				)
				, array(
					'GETvar'	=> 'tx_srfeuserregister_pi1[sword]'
				)
				, array(
					'GETvar'	=> 'tx_srfeuserregister_pi1[sort]'
				)
			)
			, 'scal'	=> array(
				array(
					'GETvar'	=> 'tx_desimplecalendar_pi1[showUid]'
				)
				, array(
					'GETvar'	=> 'tx_desimplecalendar_pi1[form]'
				)
				, array(
					'GETvar'	=> 'tx_desimplecalendar_pi1[mode]'
				)
				, array(
					'GETvar'	=> 'tx_desimplecalendar_pi1[backPath]'
				)
			)
			, 'calender-category'	=> array(
				array(
					'GETvar'	=> 'tx_advCalendar_pi1[category]'
				)
			)
			, 'view' => array(
				array(
					'GETvar' => 'view'
				)
			)
			, 'cforum'		=> array(
				array(
					'GETvar' => 'cat_uid'
				)
				, array(
					'GETvar' => 'conf_uid'
				)
				, array(
					'GETvar' => 'thread_uid'
				)
				, array(
					'GETvar' => 'page'
				)
				, array(
					'GETvar' => 'flag'
				)
			)
			, 'event' => array(
				array(
					'GETvar' => 'eventid'
				)
			)
			, 'ef' => array(
				array(
					'GETvar' => 'editflag'
				)
			)
			, 'start' => array(
				array(
					'GETvar' => 'start'
				)
			)
			, 'day' => array(
				array(
					'GETvar' => 'day'
				)
			)
			, 'week' => array(
				array(
					'GETvar' => 'week'
				)
			)
			, 'month' => array(
				array(
					'GETvar' => 'month'
				)
			)
			, 'bu' => array(
				array(
					'GETvar' => 'backURL'
				)
			)
			, 'cmd' => array(
				array(
					'GETvar' => 'cmd'
				)
			)
			, 'year' => array(
				array(
					'GETvar' => 'year'
				)
			)
			, 'rdfi' => array(
				array(
					'GETvar' => 'tx_nrdfimport_pi1[showUid]'
				)
			)
			, 'sponsor'	=> array(
				array(
					'GETvar'	=> 'tx_t3consultancies_pi1[showUid]'
					, 'lookUpTable' => array(
						'table' => 'tx_t3consultancies'
						, 'id_field' => 'uid'
						, 'alias_field' => 'title'
						, 'addWhereClause' => ' AND deleted != 1'
						, 'useUniqueCache' => 1
						, 'useUniqueCache_conf' => array(
							'strtolower' => 1
							, 'spaceCharacter' => '-'
						)
					)
				)
				, array(
					'GETvar'	=> 'tx_t3consultancies_pi1[service]'
				)
				, array(
					'GETvar'	=> 'tx_t3consultancies_pi1[pointer]'
				)
			)
			, 'slide-show'	=> array(
				array(
					'GETvar'	=> 'tx_gsislideshow_pi1[total]'
				)
				, array(
					'GETvar'	=> 'tx_gsislideshow_pi1[lastUid]'
				)
				, array(
					'GETvar'	=> 'tx_gsislideshow_pi1[firstUid]'
				)
				, array(
					'GETvar'	=> 'tx_gsislideshow_pi1[current]'
				)
				, array(
					'GETvar'	=> 'tx_gsislideshow_pi1[showUid]'
					, 'lookUpTable' => array(
						'table' => 'tx_gsislideshow_images'
						, 'id_field' => 'uid'
						, 'alias_field' => 'caption'
						, 'addWhereClause' => ' AND deleted != 1'
						, 'useUniqueCache' => 1
						, 'useUniqueCache_conf' => array(
							'strtolower' => 1
							, 'spaceCharacter' => '-'
						)
					)
				)
			)
			, 'tac' => array(
				array(
					'GETvar' => 'tac'
				)
			)
			, 'bp' => array(
				array(
					'GETvar' => 'backPID'
				)
			)
			, 'product' => array(
				array(
					'GETvar' => 'tt_products'
					, 'lookUpTable' => array(
						'table' => 'tt_products'
						, 'id_field' => 'uid'
						// MLC Goolge wants uniqueness for spidering
						, 'alias_field' => 'concat(title, " ", uid)'
						, 'addWhereClause' => ' AND deleted != 1'
						, 'useUniqueCache' => 1
						, 'useUniqueCache_conf' => array(
							'strtolower' => 1
						),
					)
				)
			)
			// MLC Bahag photo gallery
			, 'gallery' => array(
				array(
					'GETvar' => 'gallery'
				)
			)
			, 'image' => array(
				array(
					'GETvar' => 'viewImage'
				)
			)
			, 'rp' => array(
				array(
					'GETvar' => 'resultPage'
				)
			)
			, 'idx' => array(
				array(
					'GETvar' => 'idx'
				)
			)
			, 'anmode' => array (
					array('GETvar' => 'tx_piapappnote_pi1[mode]')
			)
			, 'anptr' => array (
					array('GETvar' => 'tx_piapappnote_pi1[pointer]')
)
			, 'anfile' => array (
					array('GETvar' => 'tx_piapappnote_pi1[file]')
			)
			, 'anseach' => array (
					array('GETvar' => 'tx_piapappnote_pi1[sword]')
			)
			, 'annote' => array (
					array('GETvar' => 'tx_piapappnote_pi1[noteid]')
			)
			, 'anauth' => array (
					array('GETvar' => 'tx_piapappnote_pi1[author]')
			)
			, 'anname' => array (
					array('GETvar' => 'tx_piapappnote_pi1[title]')
			)
			, 'andesc' => array (
					array('GETvar' => 'tx_piapappnote_pi1[description]')
			)
			, 'ancat' => array (
					array('GETvar' => 'tx_piapappnote_pi1[categorylist]')
			)
			, 'anver' => array (
					array('GETvar' => 'tx_piapappnote_pi1[versionlist]')
			)
			, 'andev' => array (
					array('GETvar' => 'tx_piapappnote_pi1[devicelist]')
			)													   
			, 'galp' => array (
					array('GETvar' => 'tx_hldamgallery_pi1[galleryPID]')
			)													   
			, 'galcat' => array (
					array('GETvar' => 'tx_hldamgallery_pi1[galleryCID]')
			)													   
			, 'galimg' => array (
					array('GETvar' => 'tx_hldamgallery_pi1[imgID]')
			)													   
			, 'faq-category' => array (
					array('GETvar' => 'tx_irfaq_pi1[cat]'
						, 'lookUpTable' => array(
							'table' => 'tx_irfaq_cat'
							, 'id_field' => 'uid'
							, 'alias_field' => 'title'
							, 'addWhereClause' => ' AND deleted != 1'
							, 'useUniqueCache' => 1
							, 'useUniqueCache_conf' => array(
								'strtolower' => 1
								, 'spaceCharacter' => '-'
							)
					)
				)
			)													   
			// page comments
			, 'skcomm' => array(
					array(
						'GETvar' => 'tx_skpagecomments_pi1[showComments]',
					),
					array(
						'GETvar' => 'tx_skpagecomments_pi1[showForm]',
					),
			)
			// ab_downloads
			, 'dl-act' => array(
				array(
					'GETvar' => 'tx_abdownloads_pi1[action]',
					'valueMap' => array(
						'show-category' => 'getviewcategory',
						'propose-a-new-download' => 'getviewaddnewdownload', 
						'open-download' => 'getviewclickeddownload', 
						'show-details-for-download' => 'getviewdetailsfordownload', 
						'report-broken-download' => 'getviewreportbrokendownload',
						'rate-download' => 'getviewratedownload',
					),
				),
			)
			, 'dl-cat' => array(
				array(
					'GETvar' => 'tx_abdownloads_pi1[category_uid]',
					'valueMap' => array(
						'home' => '0',
					),
					'lookUpTable' => array(
						'table' => 'tx_abdownloads_category',
						'id_field' => 'uid',
						'alias_field' => 'label',
						'addWhereClause' => ' AND deleted != 1',
						'useUniqueCache' => 1,
						'useUniqueCache_conf' => array(
							'strtolower' => 1,
							'spaceCharacter' => '-',
						),
					),
				),
			)
			, 'dl-file' => array(
				array(
					'GETvar' => 'tx_abdownloads_pi1[uid]',
					'lookUpTable' => array(
						'table' => 'tx_abdownloads_download',
						'id_field' => 'uid',
						'alias_field' => 'label',
						'addWhereClause' => ' AND deleted != 1',
						'useUniqueCache' => 1,
						'useUniqueCache_conf' => array(
							'strtolower' => 1,
							'spaceCharacter' => '-',
						),
					),
				),
			)
			, 'dl-ptr' => array(
				array(
					'GETvar' => 'tx_abdownloads_pi1[pointer]',
				),
			)
			, 'll-act' => array(
				array(
					'GETvar' => 'tx_ablinklist_pi1[action]',
					'valueMap' => array(
						'show-category' => 'getviewcategory',
						'propose-a-new-link' => 'getviewaddnewlink', 
						'open-link' => 'getviewclickedlink', 
						'show-details-for-link' => 'getviewdetailsforlink', 
						'report-broken-link' => 'getviewreportbrokenlink',
						'rate-link' => 'getviewratelink',
					),
				),
			)
			, 'll-cat' => array(
				array(
					'GETvar' => 'tx_ablinklist_pi1[category_uid]',
					'valueMap' => array(
						'home' => '0',
					),
					'lookUpTable' => array(
						'table' => 'tx_ablinklist_category',
						'id_field' => 'uid',
						'alias_field' => 'label',
						'addWhereClause' => ' AND deleted != 1',
						'useUniqueCache' => 1,
						'useUniqueCache_conf' => array(
							'strtolower' => 1,
							'spaceCharacter' => '-',
						),
					),
				),
			)
			, 'll-link' => array(
				array(
					'GETvar' => 'tx_ablinklist_pi1[uid]',
					'lookUpTable' => array(
						'table' => 'tx_ablinklist_link',
						'id_field' => 'uid',
						'alias_field' => 'label',
						'addWhereClause' => ' AND deleted != 1',
						'useUniqueCache' => 1,
						'useUniqueCache_conf' => array(
							'strtolower' => 1,
							'spaceCharacter' => '-',
						),
					),
				),
			)
			, 'll-ptr' => array(
				array(
					'GETvar' => 'tx_ablinklist_pi1[pointer]',
				),
			)
			, 'cal'=> array(
				array(
					'GETvar' => 'tx_cal_controller[view]'
				),
				array(
					'GETvar' => 'tx_cal_controller[getdate]'
				),
				array(
					'GETvar' => 'tx_cal_controller[lastview]'
				),
				array(
					'GETvar' => 'tx_cal_controller[type]'
				),
				array(
					'GETvar' => 'tx_cal_controller[category]',
					'lookUpTable' => array(
						'table' => 'tx_cal_category',
						'id_field' => 'uid',
						'alias_field' => 'title',
						'addWhereClause'  => ' AND deleted != 1',
						'useUniqueCache' => 1,
						'useUniqueCache_conf' => array(
							'strtolower' => 1,
							'spaceCharacter' => '-',
						 ),
					 
					),
				),
				array(
					'GETvar' => 'tx_cal_controller[uid]',
					'lookUpTable' => array(
						'table' => 'tx_cal_event',
						'id_field' => 'uid',
						'alias_field' => 'title',
						'addWhereClause'  => ' AND deleted != 1',
						'useUniqueCache' => 1,
						'useUniqueCache_conf' => array(
							'strtolower' => 1,
							'spaceCharacter' => '-',
						),
					),
				),
			)
			, 'ch' => array(
				array(
					'GETvar' => 'cHash',
					'noMatch' => 'bypass',
				),
			),
		)
	)
	, 'fileName' => array (
		'index' => array(
			'index.html' => array(
				'keyValues' => array(
					'type' => 0,
				)
			)
			, 'print.html' => array(
				'keyValues' => array(
					'type' => 98,
				)
			)
			, 'text.html' => array(
				'keyValues' => array(
					'type' => 99,
				)
			)
			, 'rss.xml' => array(
				'keyValues' => array(
					'type' => 100,
				)
			)
			, 'rss091.xml' => array(
				'keyValues' => array(
					'type' => 101,
				 )
			)
			, 'rdf.xml' => array(
				'keyValues' => array(
					'type' => 102,
				)
			)
			, 'atom.xml' => array(
				'keyValues' => array(
					'type' => 103,
				)
			)
			// ext:seo_basics overrides this
			, 'sitemap.xml' => array(
				'keyValues' => array(
					'type' => 776,
				)
			)
			, '_DEFAULT' => array(
				'keyValues' => array()
			)
		)
		, 'defaultToHTMLsuffixOnPrev' => 1
		, 'acceptHTMLsuffix' => 1
	)   
);

$customFile						= '../../realurl-custom.php';

// include customizations outside of this script
if ( false && file_exists( $customFile ) )
{
	include( $customFile );
}

// phpsuexec seems to not like the above, therefore put your realurl
// customizations here. don't forget to backup this file outside of this
// directory
elseif ( false )
{
	$TYPO3_CONF_VARS['EXTCONF']['realurl']['www.cannonbose.com'] = $TYPO3_CONF_VARS['EXTCONF']['realurl']['_DEFAULT'];
	$TYPO3_CONF_VARS['EXTCONF']['realurl']['www.cannonbose.com']['pagePath']['rootpage_id'] = 1;
	$TYPO3_CONF_VARS['EXTCONF']['realurl']['cannonbose.com'] = $TYPO3_CONF_VARS['EXTCONF']['realurl']['www.cannonbose.com'];
	$TYPO3_CONF_VARS['EXTCONF']['realurl']['support.cannonbose.com'] = $TYPO3_CONF_VARS['EXTCONF']['realurl']['_DEFAULT'];
	$TYPO3_CONF_VARS['EXTCONF']['realurl']['support.cannonbose.com']['pagePath']['rootpage_id'] = 7;
}

?>


realurl-custom-sample.php
<?php

/**
 * RealURL alternate rootpage_id and multiple domain setup example
 *
 * multiple domain setup
 * 	Update domains
 * 	Update rootpage_id
 *	The root page id is the uid of your domain home in the TYPO3 page tree
 *
 * @author Michael Cannon <michael@cannonbose.com>
 * @version $Id: realurl-custom-sample.php,v 1.1 2008/04/02 19:57:18 cannon Exp $
 */


// Edit rootpage_id to your website's root page UID
if ( false )
{
	$TYPO3_CONF_VARS['EXTCONF']['realurl']['_DEFAULT']['pagePath']['rootpage_id'] = 1;
}

// edit for single subdomains
if ( false )
{
	$TYPO3_CONF_VARS['EXTCONF']['realurl']['subdomain.example.com'] = $TYPO3_CONF_VARS['EXTCONF']['realurl']['_DEFAULT'];
	$TYPO3_CONF_VARS['EXTCONF']['realurl']['subdomain.example.com']['pagePath']['rootpage_id'] = 357;
}

// edit for multiple domains
if ( false )
{
	$TYPO3_CONF_VARS['EXTCONF']['realurl']['www.example.com'] = $TYPO3_CONF_VARS['EXTCONF']['realurl']['_DEFAULT'];
	$TYPO3_CONF_VARS['EXTCONF']['realurl']['www.example.com']['pagePath']['rootpage_id'] = 45;
	$TYPO3_CONF_VARS['EXTCONF']['realurl']['example.com'] = $TYPO3_CONF_VARS['EXTCONF']['realurl']['www.example.com'];
}

?>


ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext: "cbrealurl"
#
# Auto generated 02-04-2008 18:49
#
# Manual updates:
# Only the data in the array - anything else is removed by next write.
# "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'Cannonbose RealURL Configuration Helper',
	'description' => 'Pre-configured RealURL for many extensions and multiple websites. Read README.txt for directions.',
	'category' => 'misc',
	'author' => 'Michael Cannon',
	'author_email' => 'michael@cannonbose.com',
	'shy' => '',
	'dependencies' => 'realurl',
	'conflicts' => '',
	'priority' => '',
	'module' => '',
	'state' => 'alpha',
	'internal' => '',
	'uploadfolder' => 0,
	'createDirs' => '',
	'modify_tables' => '',
	'clearCacheOnLoad' => 0,
	'lockType' => '',
	'author_company' => '',
	'version' => '1.1.0',
	'constraints' => array(
		'depends' => array(
			'realurl' => '',
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:12:{s:9:"CHANGELOG";s:4:"3eca";s:10:"README.txt";s:4:"580d";s:12:"ext_icon.gif";s:4:"b328";s:17:"ext_localconf.php";s:4:"ad6d";s:14:"ext_tables.php";s:4:"6b35";s:25:"realurl-custom-sample.php";s:4:"0a6b";s:11:"realurl.php";s:4:"ba37";s:31:"static/CB_RealURL/constants.txt";s:4:"0996";s:27:"static/CB_RealURL/setup.txt";s:4:"d0ae";s:14:"doc/manual.sxw";s:4:"d95b";s:19:"doc/wizard_form.dat";s:4:"5371";s:20:"doc/wizard_form.html";s:4:"a766";}',
	'suggests' => array(
	),
);

?>